pkg/openservicebroker/cmd/server/start.go

```diff
@@ -0,0 +1,105 @@
+package server
+
+import (
+	"fmt"
+	"io"
+	"net"
+
+	"github.com/spf13/cobra"
+
+	genericapiserver "k8s.io/apiserver/pkg/server"
+	genericoptions "k8s.io/apiserver/pkg/server/options"
+
+	"github.com/openshift/origin/pkg/openservicebroker/server"
+)
+
+const defaultEtcdPathPrefix = "/registry/templateservicebroker.openshift.io"
+
+type TemplateServiceBrokerServerOptions struct {
+	RecommendedOptions *genericoptions.RecommendedOptions
+
+	StdOut io.Writer
+	StdErr io.Writer
+
+	TemplateNamespaces []string
+}
+
+func NewTemplateServiceBrokerServerOptions(out, errOut io.Writer) *TemplateServiceBrokerServerOptions {
+	o := &TemplateServiceBrokerServerOptions{
+		RecommendedOptions: genericoptions.NewRecommendedOptions(defaultEtcdPathPrefix, server.Scheme, server.Codecs.LegacyCodec()),
+
+		StdOut: out,
+		StdErr: errOut,
+	}
+
+	return o
+}
+
+func NewCommandStartTemplateServiceBrokerServer(out, errOut io.Writer, stopCh <-chan struct{}) *cobra.Command {
+	o := NewTemplateServiceBrokerServerOptions(out, errOut)
+
+	cmd := &cobra.Command{
+		Use:   "template-service-broker",
+		Short: "Launch a template service broker server",
+		Long:  "Launch a template service broker server",
+		RunE: func(c *cobra.Command, args []string) error {
+			if err := o.Complete(); err != nil {
+				return err
+			}
+			if err := o.Validate(args); err != nil {
+				return err
+			}
+			if err := o.RunTemplateServiceBrokerServer(stopCh); err != nil {
+				return err
+			}
+			return nil
+		},
+	}
+
+	flags := cmd.Flags()
+	o.RecommendedOptions.AddFlags(flags)
+	flags.StringSliceVar(&o.TemplateNamespaces, "template-namespace", o.TemplateNamespaces, "TemplateNamespaces indicates the namespace(s) in which the template service broker looks for templates to serve to the catalog.")
+
+	return cmd
+}
+
+func (o TemplateServiceBrokerServerOptions) Validate(args []string) error {
+	return nil
+}
+
+func (o *TemplateServiceBrokerServerOptions) Complete() error {
+	return nil
+}
+
+func (o TemplateServiceBrokerServerOptions) Config() (*server.TemplateServiceBrokerConfig, error) {
+	// TODO have a "real" external address
+	if err := o.RecommendedOptions.SecureServing.MaybeDefaultWithSelfSignedCerts("localhost", nil, []net.IP{net.ParseIP("127.0.0.1")}); err != nil {
```

Comment:
at what point is this going to serve with a cert signed by the cluster CA?  Isn't that going to be necessary for the SC to cleanly talk to it?